src/main/java/com/exasol/adapter/dynamodb/mapping/AbstractColumnMappingDefinition.java

```diff
@@ -0,0 +1,134 @@
+ */
+public abstract class AbstractColumnMappingDefinition implements Serializable {
+    private static final long serialVersionUID = 48342992735371252L;
+    private final String exasolName; // name of the column in resulting Virtual Schema
+    private final AbstractDynamodbResultWalker pathToSourceProperty;
+    private final LookupFailBehaviour lookupFailBehaviour;
+
+    /**
+     * Creates an instance of {@link AbstractColumnMappingDefinition}.
+     * 
+     * @param parameters parameter object
+     */
+    public AbstractColumnMappingDefinition(final ConstructorParameters parameters) {
+        this.exasolName = parameters.destinationName;
+        this.pathToSourceProperty = parameters.pathToSourceProperty;
+        this.lookupFailBehaviour = parameters.lookupFailBehaviour;
+    }
+
+    /**
+     * Get the name of the column in the Exasol table.
+     *
+     * @return name of the column
+     */
+    public String getExasolName() {
+        return this.exasolName;
+    }
+
+    /**
+     * Get the Exasol data type.
+     *
+     * @return Exasol data type
+     */
+    public abstract DataType getExasolDataType();
+
+    /**
+     * Get the default value of this column.
+     *
+     * @return {@link ValueExpression} holding default value
+     */
+    public abstract ValueExpression getExasolDefaultValue();
+
+    /**
+     * Get the string representation of the exasol column default value literal.
+     * 
+     * @return default value string
+     */
+    public String getExasolDefaultValueLiteral() {
+        final StringRendererConfig stringRendererConfig = StringRendererConfig.createDefault();
+        final ValueExpressionRenderer renderer = new ValueExpressionRenderer(stringRendererConfig);
+        this.getExasolDefaultValue().accept(renderer);
+        return renderer.render();
+    }
+
+    /**
+     * Describes if Exasol column is nullable.
+     *
+     * @return {@code <true>} if Exasol column is nullable
+     */
+    public abstract boolean isExasolColumnNullable();
+
+    /**
+     * Get the {@link LookupFailBehaviour}
+     *
+     * @return {@link LookupFailBehaviour}
+     */
+    public LookupFailBehaviour getLookupFailBehaviour() {
+        return this.lookupFailBehaviour;
+    }
+
+    AbstractDynamodbResultWalker getPathToSourceProperty() {
+        return this.pathToSourceProperty;
+    }
+
+    public abstract void accept(ColumnMappingDefinitionVisitor visitor);
+
+    /**
+     * Behaviour if the requested property is not set in a given DynamoDB row.
+     */
+    public enum LookupFailBehaviour {
+        /**
+         * Break the execution of the query .
+         */
+        EXCEPTION,
+        /**
+         * The column specific default value is returned.
+         */
+        DEFAULT_VALUE
+    }
+
+    /**
+     * Parameter object for
+     * {@link AbstractColumnMappingDefinition#AbstractColumnMappingDefinition(ConstructorParameters)}
+     */
+    public static class ConstructorParameters {
```

Comment:
I need it because in the future `JsonMappingFactory` will build many different types if `ColumnMappingDefinitions`. They all have the same parameters for the common ancestor `AbstractColumnMappingDefinition`. In the Factory a method retrieves these common parameters from the JSON definition.